src/compiler/crystal/config.cr

```diff
@@ -34,7 +34,7 @@ module Crystal
     end
 
     def self.date
-      {{ env("CRYSTAL_CONFIG_BUILD_DATE") || `date "+%Y-%m-%d"`.stringify.chomp }}
+      {{ `date -u -d "@${SOURCE_DATE_EPOCH:-$(date +%s)}" +%Y-%m-%d`.stringify.chomp }}
```

Comment:
`SOURCE_DATE_EPOCH` is an integer representing the seconds since the Unix epoch. Free-flowing date formats without any spec whatsoever, on top of having 24 hours worth of lacking precision, is not very useful for storing dates, notwithstanding that crystal uses that level of imprecision in its final embedded output format.

Reply:
In any case `-d` won't work in Mac so we'll have to put an if flag condition for Linux and Mac.